src/pyclaw/sharpclaw/solver.py

```diff
@@ -307,6 +326,16 @@ def step(self,solution,take_one_step,tstart,tend):
         elif self.time_integrator == 'SSP22':
             self.ssp22(state)
 
+        elif self.time_integrator == 'DWSSP22':
+            # Implementation is in the Shu-Osher form
+            dw_euler_step = state.q - self.dt*self.dwdq_dt
```

Comment:
This should go in another register, so that memory usage is clear and to avoid creating more PETSc Vecs in parallel.